### PR TITLE
Periodic cleanup of kernel unused classes, methods, wrong javadocs and comments

### PR DESCRIPTION
**Pull request by MishaDemianenko**

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/tracing/TransactionEvent.java [hunk 1]
```diff
@@ -81,7 +81,7 @@ public void setReadOnly( boolean wasReadOnly )
 
     /**
      * Set the type of the transaction, as given by
-     * {@link org.neo4j.kernel.impl.api.KernelTransactionImplementation.TransactionType}.
+     * {@link org.neo4j.kernel.impl.api.KernelTransactionImplementation.TransactionWriteState}.
      */
     void setTransactionType( String transactionTypeName );
```
Comment (burqen) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/tracing/TransactionEvent.java]: `TransactionEvent` has no implementations that is not `no_op`. Is this part of public API? If so, why does it live in `impl`? If not, can we just delete it completely?

**Review by burqen — CHANGES_REQUESTED** [review]
Changes look good! Nice to have a little bit cleaner house 👍  Just one comment.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/tracing/TransactionEvent.java (hunk 1 above)
Comment (burqen) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/tracing/TransactionEvent.java]: Nevermind, my bad. It _has_ a real implementation.

Comment (burqen) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/tracing/TransactionEvent.java]: Although, may I suggest changing method and parameter name to reflect the change from "transactionType" to "transactionWriteState"?

Comment (MishaDemianenko) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/tracing/TransactionEvent.java]: @burqen It's not a public API, and it looks reasonable so why not. Updated.

**Review by burqen — APPROVED** [review]

**Pull request merged**